flink-table/flink-table-runtime/src/main/java/org/apache/flink/table/data/UpdatableRowData.java

```diff
@@ -17,23 +17,33 @@
 
 package org.apache.flink.table.data;
 
+import org.apache.flink.annotation.Internal;
 import org.apache.flink.table.data.binary.TypedSetters;
 import org.apache.flink.types.RowKind;
 
+import java.util.BitSet;
+
 /**
  * An implementation of {@link RowData} which is backed by a {@link RowData} and an updated Java
  * object array.
  */
+@Internal
 public final class UpdatableRowData implements RowData, TypedSetters {
 
     private RowData row;
     private final Object[] fields;
-    private final boolean[] updated;
+    private final BitSet updated;
```

Comment:
I wonder about the performance or memory impact of this. At least from a quick search it seems to be OK performance-wise and even beneficial memory-wise (https://www.baeldung.com/java-boolean-array-bitset-performance)

Reply:
You could see some actual benefit, as the number of processed rows increases, and especially if you have lots of fields select, (like in a `INSERT INTO ... SELECT * FROM ...`)